tests/broker_client/utils.go

```diff
@@ -0,0 +1,54 @@
+package broker_client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+	"encoding/json"
+)
+
+func NewServiceBrokerError(err error) *ServiceBrokerError {
+	return &ServiceBrokerError{
+		Description: err.Error(),
+	}
+}
+
+func isErrorResponse(res *http.Response) bool {
+	return res.StatusCode < http.StatusOK || res.StatusCode > http.StatusNoContent
+
+}
+
+func DoRequest(method, url, data string, obj interface{}, token, userIdentity string) (*http.Response, interface{}, *ServiceBrokerError) {
+	req, err := http.NewRequest(method, url, bytes.NewBufferString(data)); if err != nil {
+		return nil, nil, NewServiceBrokerError(err)
+	}
+
+	req.Header.Add("Content-type", "application/json")
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("Authorization", fmt.Sprintf("bearer %s", token))
+	req.Header.Add("X-Broker-API-Originating-Identity", userIdentity)
+
+	httpClient := &http.Client{
+		Timeout: time.Second * 10,
+	}
+	res, err := httpClient.Do(req);
+	defer res.Body.Close()
```

Comment:
the defer should come after the err check other wise it may be possible for the res to be nil and panic and miss the err

Reply:
:+1: